### PR TITLE
Linting For OpenAPI

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Add example for the job that validates API for correctness

## Context

http://api.appservices.tech/ defines OpenAPI standards that your team can follow to build Managed Services Control Plane APIs. We use spectral ruleset to check those rules. We will add validation into github actions once we get initial API validated and onboarded into sdk. 

## Example of objects and OpenAPI annotations to use to meet standards

This API meets all our standards
https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/pull/200

Some of the changes that were made can be copy pasted to your project etc.
Another good example of the Java/Quarkus project that follows the standard would be SRS fleet manager:

https://github.com/bf2fc6cc711aee1a0c2a/srs-fleet-manager/blob/main/core/src/main/resources/srs-fleet-manager.json#L546-L583

## How I can run this locally

http://api.appservices.tech/spectral/

### CODE REVIEW

**Review by sutaakar — APPROVED** [review]

**Review by jomarko — APPROVED** [review]

**Comment by wtrocki** [comment]
@r00ta Example of annotations that can be used from the PR mentioned above:

https://github.com/MikeEdgar/kafka-admin-api/blob/2e4b258e222d9bd54971dad0e3afca00062910df/kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/OperationsHandler.java#L34-L104

We need proper operationId, responseTypes and handling defined. 

In current form openapi file is not usable (it does require series of manual updates to pass standard openapi validation)

Let me know if we can help more on this. It would be matter of adding more annotations or changing existing ones

File: .github/actions/validate_api.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Validate API
+
+on:
+  pull_request:
+    branches: [ main, linting]
```
Comment (r00ta) [inline .github/actions/validate_api.yaml:5]: @wtrocki we don't have the `linting` branch

Comment (wtrocki) [reply .github/actions/validate_api.yaml:5]: That was used to test it on my fork. Removing

**Comment by r00ta** [comment]
Thank you @wtrocki for your contribution

**Pull request merged**